src/js/planting/planting-handlers.js

```diff
@@ -96,20 +97,37 @@ Planting.prototype.plant_objects_for_view = function () {
     };
 };
 
-var calc_direction = function calc_direction(objectPosition, mousePosition) {
+function calc_direction(objectPosition, mousePosition) {
     var calc = {};
-
-    if (objectPosition < mousePosition) {
-        calc.direction = true;
-        calc.width = mousePosition - objectPosition;
+    if (objectPosition <= mousePosition) {
+        calc = {
+            direction: true,
+            mouseObjectDiff: mousePosition - objectPosition
+        };
     } else if (objectPosition > mousePosition) {
-        calc.direction = false;
-        calc.width = objectPosition - mousePosition;
+        calc = {
+            direction: false,
+            mouseObjectDiff: objectPosition - mousePosition
+        };
     }
-
     return calc;
+}
+
+function calc_sides(mouseObjectDiff, projectionsLength) {
+    var sideLength = 20;
```

Comment:
Jeżeli to jest stała wartość to w jakichś globalnych stałych powinniśmy to mieć.